code/modules/reagents/chemistry/reagents/blob_reagents.dm

```diff
@@ -236,8 +235,8 @@
 	if(M)
 		M.apply_damage(0.6*reac_volume, OXY)
 
-/datum/reagent/blob/energized_jelly/damage_reaction(obj/structure/blob/B, original_health, damage, damage_type, cause)
-	if(!isnull(cause) && original_health - damage <= 0 && prob(10))
+/datum/reagent/blob/energized_jelly/damage_reaction(obj/structure/blob/B, damage, damage_type, damage_flag)
+	if((damage_flag == "melee" || damage_flag == "bullet" || damage_flag == "laser") && B.obj_integrity - damage <= 0 && prob(10))
 		spark_system.set_up(rand(2, 4), 0, B)
 		spark_system.start()
 	return ..()
```

Comment:
The B.take_damage() line for the emp_reaction for this chemical should use the "energy" armor flag.